src/main/groovy/life/qbic/samplestatus/reporter/services/users/UserService.groovy

```diff
@@ -0,0 +1,19 @@
+package life.qbic.samplestatus.reporter.services.users
+
+import life.qbic.samplestatus.reporter.api.UserDetails
+
+/**
+ * <b>Provides user information</b>
+ *
+ * @since 1.0.0
+ */
+interface UserService {
+    /**
+     * Retrieves user details for a provided userId from the persistence layer.
+     * @param userId the identifier of the user
+     * @return user details for a user found.
+     * @since 1.0.0
+     */
+    UserDetails getUserDetails(String userId)
```

Comment:
What happens if no user details can be retrieved for a given id? I think the API can profit from returning an `Optional<UserDetails>`.